### PR TITLE
feat(): add examples

### PR DESCRIPTION
**Pull request by rluvaton**
add function to add multiple examples at once

#1683

### CODE REVIEW

**Comment by bcoe** [comment]
I like this idea 👍 I think I would rather we overrode `example` to accept a `string` or `string[]` rather than adding the new `.examples()` though -- there are already several methods that accept both the singular and array form.

Thank you for the contribution.

**Comment by rluvaton** [comment]
> I like this idea 👍 I think I would rather we overrode `example` to accept a `string` or `string[]` rather than adding the new `.examples()` though -- there are already several methods that accept both the singular and array form.
> 
> 
> 
> Thank you for the contribution.

Fixed it in the latest commit
_(For some reason this PR hasn’t updated with that commit 😬)_

**Comment by rluvaton** [comment]
@bcoe I recommend you would squash this PR (it contain lots _oh I forgot something_ commit 😬)

File: docs/api.md [hunk 1]
```diff
@@ -766,12 +766,23 @@ var argv = require('yargs')
 
 .example(cmd, desc)
 -------------------
+.example(exampleArray)
```
Comment (bcoe) [inline docs/api.md:769]:
```suggestion
.example([[cmd1, desc1], [cmd2, desc2]])
```

☝️  what do you think?

**Comment by bcoe** [comment]
> I recommend you would squash this PR (it contain lots oh I forgot something commit 😬)

@rluvaton no worries, we always squash 😄

File: docs/api.md (hunk 1 above)
Comment (rluvaton) [reply docs/api.md:769]:
```suggestion

.example([[cmd1, desc1], [cmd2, desc2], ...])
```

Maybe that (added 3 dots to let them know it can be any size)?

**Comment by bcoe** [comment]
@rluvaton thank you for the contribution.

**Comment by rluvaton** [comment]
@bcoe when this feature should be release?

**Comment by bcoe** [comment]
@rluvaton if you have a few moments, mind testing out a pre-release of this feature:

```
npm i yargs@15.4.0-beta.1
```

I'll release soon if things are looking good.

**Comment by rluvaton** [comment]
> @rluvaton if you have a few moments, mind testing out a pre-release of this feature:
> 
> ```
> npm i yargs@15.4.0-beta.1
> ```
> 
> I'll release soon if things are looking good.

I tried it and it looks good so far (except the missing type definition for the new feature [obviously])

**Comment by bcoe** [comment]
@rluvaton okay, `yargs@15.4.0` is actually released.

**Pull request merged**